tools/fv_eta.F90

```diff
@@ -351,6 +351,26 @@ subroutine set_eta(km, ks, ptop, ak, bk, npz_type)
          auto_routine = 2
       end select
 
+! Jili Dong add ak/bk input
+   else if (trim(npz_type) == 'input') then
+       open (113,file="global_hyblev_fcst.txt")
+       read(113,*)
+       do k=km+1,1,-1
+          read(113,*) ak(k),bk(k)
+       end do
+       close(113)
+
+       do k=1,km+1
+          if (bk(k) .eq. 0.0) then
+             ks=k
+             exit
+          end if
+       end do
+
+       ks=ks-2
```

Comment:
@lharris4 - please confirm that ks-2 is correct for the point where bk becomes zero.  I thought it should be ks-1 to go from levels to layers.

Reply:
it appears the easiest way to get ks is through calling set_external_eta. I will replace this part with:

call set_external_eta(ak, bk, ptop, ks)